publisher.rb

```diff
@@ -0,0 +1,47 @@
+require 'aws-sdk-sqs'
+require 'byebug'
+require_relative 'load_aws'
+
+class Publisher
+  # Note: You can store the values of the aws access keys and region list in a config file (json) 
+  # from where you can get the values into the constants.
+  ACCESS_KEY_ID = AWS["access_key_id"]
+  SECRET_ACCESS_KEY = AWS["secret_access_key"]
+  REGION = AWS["region"]
+
+  attr_accessor :sqs
+
+  def initialize
+    creds = Aws::Credentials.new(ACCESS_KEY_ID, SECRET_ACCESS_KEY)
+    @sqs = Aws::SQS::Client.new(region: REGION, credentials: creds)
+  end
+
+  # For Standard Queue
+  def create_std_q(queue_name)
+    std_queue = sqs.create_queue(queue_name: queue_name)
+  end
+
+  def publish_messages(queue_url, options=[])
+    delay_seconds = options[1].nil? ? 5 : options[1]
```

Comment:
Put a check to ensure that no message is published less than 5 seconds apart. Don't want you to be charged un-necessarily.
So 5 or larger

Reply:
@glen In that case, we will have to use sleep(). 
AWS SQS delay_seconds basically ensures that the same message is not sent within the specified delay_time. This is particularly for standard queues where the same message from different servers can be sent multiple times. Standard queues don't maintain a unique group id or deduplication id like the FIFO counter parts. So the same message replicated on multiple SQS servers may be sent multiple times or multiple attempts be made to send them. This thing takes care of that. If we want to generate and send messages 5 or whatever seconds apart, will have to use sleep().